memory/ndctl.py

```diff
@@ -163,21 +205,37 @@ def setUp(self):
                         self.smm.install(pkg):
                     self.cancel('%s is needed for the test to be run' % pkg)
 
+            # Using fetch_asset() API
+            '''
             git_branch = self.params.get('git_branch', default='pending')
-            location = "https://github.com/pmem/ndctl/archive/"
-            location = location + git_branch + ".zip"
-            tarball = self.fetch_asset("ndctl.zip", locations=location,
-                                       expire='7d')
+            #location = location + git_branch + ".zip"
+            location = "https://github.com/pmem/ndctl/archive/refs/heads/pending.zip"
+            tarball = self.fetch_asset("ndctl.zip", locations=location, expire='7d')
             archive.extract(tarball, self.teststmpdir)
             os.chdir("%s/ndctl-%s" % (self.teststmpdir, git_branch))
-            process.run('./autogen.sh', sudo=True, shell=True)
-            process.run("./configure CFLAGS='-g -O2' --prefix=/usr "
-                        "--disable-docs "
-                        "--sysconfdir=/etc --libdir="
-                        "/usr/lib64", shell=True, sudo=True)
-            build.make(".")
-            self.ndctl = os.path.abspath('./ndctl/ndctl')
-            self.daxctl = os.path.abspath('./daxctl/daxctl')
+            '''
+
+            # Using get_repo() API
+            # Clone the branch
+            git.get_repo(url, branch='pending', destination_dir=self.teststmpdir)
```

Comment:
please make use of yaml input

Reply:
Are you pointing to the '_url_' variable used in the get_repo API??

Reply:
looking for the branch as here we are hardcoding

Reply:
okay got it!